engine/src/main/java/graphql/nadel/engine/execution/Execution.java

```diff
@@ -163,10 +163,17 @@ private static NadelInstrumentationExecuteOperationParameters buildInstrumentati
     private FieldInfos createFieldsInfos() {
         Map<GraphQLFieldDefinition, FieldInfo> fieldInfoByDefinition = new LinkedHashMap<>();
 
+        GraphQLObjectType schemaQueryType = overallSchema.getQueryType();
```

Comment:
I don't think this is really needed.
If the goal of `FieldInfo` is only to map fields to services and the fields that use dynamic service resolution are not part of any particular one service, there's no point in creating `FieldInfo` objects for them.

Reply:
Nominally this should be done for Mutation type as well.

Reply:
I ultimately killed this code as it achieves nothing.